webui/react/src/pages/TrialDetails.tsx

```diff
@@ -140,7 +140,7 @@ const TrialDetailsComp: React.FC = () => {
 
   const columns = useMemo(() => {
     const checkpointRenderer = (_: string, record: Step) => {
-      if (record.checkpoint) {
+      if (record.checkpoint && record.checkpoint.state !== CheckpointState.Deleted) {
```

Comment:
nit: suggestion: capture this repeated bit a `hasCheckpoint` helper

Reply:
Done, I'll merge when you lmk is good.

Reply:
thanks for updating this :rocket: :ship: